CSquared/Evaluator/Evaluables/Statements/VariableAssignment.cs

```diff
@@ -6,37 +6,47 @@
 
 namespace CSquared
 {
-    public partial class VariableAssignment : IStatement
-    {
-        public IExpression Evaluate(CSquaredEnvironment environment)
-        {
-            var variableIdentifier = this.Identifier.Value.ToString();
-
-            var expression =
-                 this.Expression.IsNull() ?
-                     new Null() :
-                     this.Expression.Evaluate(environment);
-
-            
-            if (!this.IndexExpression.IsNull())
-            {
-                IExpression indexableExpression = null;
-
-                if (!environment.Lookup(variableIdentifier, out indexableExpression))
-                {
-                    throw new Exception("Attempted to assign value to undeclared variable " + variableIdentifier);
-                }
-
-                var indexable = indexableExpression.CastTo<IIndexable>();
-
-                indexable.SetIndex(environment, this.IndexExpression, expression);
-            }
-            else if (!environment.Update(variableIdentifier, expression))
-            {
-                throw new Exception("Attempted to assign value to undeclared variable " + variableIdentifier);
-            }
-
-            return new Null();
-        }
-    }
+	public partial class VariableAssignment : IStatement
+	{
+		public IExpression Evaluate(CSquaredEnvironment environment)
+		{
+			var variableIdentifier = this.Identifier.Value.ToString();
+
+			var expression =
+				 this.Expression.IsNull() ?
+					 new Null() :
+					 this.Expression.Evaluate(environment);
+
+
+			if (!this.IndexExpression.IsNull())
+			{
+				IExpression indexableExpression = null;
+
+				if (!environment.Lookup(variableIdentifier, out indexableExpression))
+				{
+					throw new Exception("Attempted to assign value to undeclared variable " + variableIdentifier);
+				}
+
+				var indexable = indexableExpression.CastTo<IIndexable>();
+
+				indexable.SetIndex(environment, this.IndexExpression, expression);
+			}
+			else if (!environment.Update(variableIdentifier, expression))
+			{
+				var variableDeclaration = new VariableDeclaration
+				{
+					IdentifierDeclaration = new IdentifierDeclaration
+					{
+						Identifier = this.Identifier,
+						ScopeEscalated = false
+					},
+					Expression = this.Expression
+				};
+
+				return variableDeclaration.Evaluate(environment);
+			}
+
+			return new Null();
+		}
+	}
```

Comment:
[fix spacing. tabs should be four spaces](http://stackoverflow.com/questions/5755942/how-do-you-auto-format-code-in-visual-studio). it's throwing off the diff, so it's hard to tell exactly what changes you made..